### DISCUSSION TITLE
How to custom chart(chart host) size and let it to fit in a fixed size of box or frame?

### DISCUSSION THREAD

**Discussion by siyingch86** [patrykandpatrick/vico#604] [Questions] [answered]
Hi, I have a question regarding chart size. For example, the chart below is in your example. It is scrollable. But what if i want it to not scroll, and let every bar spacing automatically shrink and show all the bars within the screen width. Is there any way to do that?        
<img width="370" alt="Screenshot 2024-03-06 at 9 26 05 PM" src="https://github.com/patrykandpatrick/vico/assets/160801134/9a050f6f-b3d9-432e-ad62-e658deeee602">

**Comment by patrickmichalik** [comment]
Hello! Yup, this is possible. Are you using Vico 1 or Vico 2, and are you using it with Jetpack Compose or with the view system?

**Reply by siyingch86** [reply]
I'm using vico2 with Compose

**Reply by patrickmichalik** [reply, marked as answer]
Got it. You can use the `scrollEnabled` parameter of `rememberVicoScrollState` to turn off scroll. `CartesianChartHost` accepts a `VicoScrollState` instance. The zoom settings will be adapted automatically.

The `HorizontalAxis` may get overly compressed, in which case it may be a good idea to switch to `HorizontalLayout.FullWidth` and use `AxisItemPlacer.Horizontal.default`, setting `addExtremeLabelPadding` to `true`.
